app/components/find/results/results_component.rb

```diff
@@ -5,12 +5,13 @@ module Results
     class ResultsComponent < ViewComponent::Base
       include ::ViewHelper
 
-      attr_reader :results, :courses
+      attr_reader :results, :courses, :pagy
 
-      def initialize(results:, courses:)
+      def initialize(results:, courses:, pagy:)
```

Comment:
Must say, I find it a bit concerning passing pagy through Components. What if there are 3 levels of nested components?

Not for now, but something to consider.

Reply:
For those who have used react, they'll recognise this problem

Reply:
Yes, I agree. 

In Apply, we previously used a `PaginatorComponent` to encapsulate pagination logic (at least on the interface). With Kaminari that is easy. 

While Pagy is a powerful gem, its unique interface do differ significantly from Active Record relation & scopes, which makes it a bit challenging to integrate in an abstraction like `PaginatorComponent` or some sorts.